src/java/org/apache/cassandra/db/marshal/AbstractType.java

```diff
@@ -642,7 +722,11 @@ public final boolean referencesUserTypes()
      */
     public AbstractType<?> withUpdatedUserType(UserType udt)
     {
-        return this;
+        // Note that the UDT must override this method.
```

Comment:
Maybe we want to add an assertion like the one in `referencesUserType`, or just do nothing, but we should probably use the same criteria.